hector_pose_prediction_ros/include/hector_pose_prediction_ros/visualization.h

```diff
@@ -0,0 +1,226 @@
+    marker.scale.y = 0.02; // head diameter
+    marker.scale.z = 0.01; // if not 0, head length
+
+    marker.header.frame_id = frame_id;
+    marker.ns = ns;
+    marker.id = static_cast<int32_t>( i );
+
+    // Identity
+    marker.pose.orientation.w = 1.0;
+
+    geometry_msgs::Point point_start_msg;
+    tf::pointEigenToMsg( contact_points[i].template cast<double>(), point_start_msg );
+    marker.points.push_back( point_start_msg );
+
+    Eigen::Vector3d point_end = contact_points[i].template cast<double>() +
+                                normals_scale * normals[i].template cast<double>();
+    geometry_msgs::Point point_end_msg;
+    tf::pointEigenToMsg( point_end, point_end_msg );
+    marker.points.push_back( point_end_msg );
+
+    // Determine color based on angle between normal and (inverted) gravity vector
+    hector_math::Vector3<Scalar> gravity_vector = hector_math::Vector3<Scalar>::UnitZ();
+    Scalar angle = std::acos( gravity_vector.dot( normals[i] ) / ( normals[i].norm() ) );
+    auto v = static_cast<float>( std::max( -2 / M_PI * angle + 1, 0.0 ) ); // Value between 1 and 0
+    marker.color.a = 1.0f;
+    marker.color.r = 1.0f - v;
+    marker.color.g = v;
+    marker.color.b = 0.0f;
+
+    marker_array.markers.push_back( marker );
+  }
+}
+
+// Convenience functions
+
+template<typename Scalar>
+void drawFullSupportPolygon(
+    visualization_msgs::MarkerArray &marker_array,
+    const hector_pose_prediction_interface::SupportPolygon<Scalar> &support_polygon,
+    const std::string &frame_id )
+{
+  // Draw Polygon with stability if available
+  if ( support_polygon.edge_stabilities.empty() ) {
+    drawSupportPolygon( marker_array, support_polygon.contact_hull_points,
+                        Eigen::Vector4f( 0.0f, 1.0f, 0.0f, 1.0f ), frame_id, "support_polygon" );
+  } else {
+    drawSupportPolygonWithStability( marker_array, support_polygon.contact_hull_points,
+                                     support_polygon.edge_stabilities, 6.0, frame_id,
+                                     "support_polygon" );
+  }
+  // Draw convex hull points
+  drawContactPoints( marker_array, support_polygon.contact_hull_points,
+                     Eigen::Vector4f( 0, 1, 0, 1 ), frame_id, "support_polygon_points", 0.04 );
+}
+
+template<typename Scalar>
+void drawFullContactInformation(
+    visualization_msgs::MarkerArray &marker_array,
+    const hector_pose_prediction_interface::ContactInformation<Scalar> &contact_information,
+    const std::string &frame_id )
+{
+  // Collect data
+  hector_math::Vector3List<Scalar> contact_points;
+  contact_points.reserve( contact_information.contact_points.size() );
+  hector_math::Vector3List<Scalar> normals;
+  normals.reserve( contact_information.contact_points.size() );
+  for ( const auto &contact_point : contact_information.contact_points ) {
+    contact_points.push_back( contact_point.point );
+    normals.push_back( contact_point.surface_normal );
+  }
+  // Draw contact points
```

Comment:
This is a good example of a bad comment since it only uses space and adds no information.
Don't add comments for the sake of having comments. The code is written well enough to be easily understood.

Reply:
I generally try to add comments that make it easier for me to understand the code later. I revised the comments a bit, but the remaining ones are at least helpful to me.

Reply:
Yeah most made sense, I just found it funny to have a comment that is literally the same as the name of the function call in the next line :P